### ISSUE TITLE
Allow to mark the language names as translation strings

### ISSUE THREAD

**Issue by murray3k** [wagtail/wagtail#6571]
### Is your proposal related to a problem?

It would be nice to have the names of all used languages and therefore also the names of all Locales to be "localized".
Being displayed as "Français" for "French" or "Deutsch" for "German" and so on.

### Describe the solution you'd like

In the [Django Docs](https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#how-django-discovers-language-preference) this is already covered:

```python
from django.utils.translation import gettext_lazy as _

LANGUAGES = [
       ('de', _('German')),
       ('en', _('English')),
]
```
It would be nice if this could "just work". 😉 

### Additional context

Doing just that in `WAGTAIL_CONTENT_LANGUAGES` and `LANGUAGES` breaks the rendering of the Admin-Interface and also the frontend, if something like `{{ page.locale }}` is used, throwing the infamous `__str__ returned non-string (type __proxy__)`.

A solution is simple, everything seems to work fine just doing this:
```python
class Locale(models.Model):
    ...
    def __str__(self):
        return force_str(self.get_display_name() or self.language_code)
```

I'll of course provide a PR, but I don't know how to test for this. Just putting `_(...)` here and there passes tests just fine, but is it enough? Any ideas?

Thanks.

**Comment by Shaffle1** [comment]
+1
I spend 2 or 3 hours fixing this, I didn't notice when this `__str__ returned non-string (type __proxy__)` error started to appear 🤦‍♂️. I also think it would be nice to have 😄.

**Comment by fordguo** [comment]
Thanks. I get the same problem.

**Comment by kaedroho** [comment]
Thanks for reporting!

> I'll of course provide a PR, but I don't know how to test for this. Just putting _(...) here and there passes tests just fine, but is it enough? Any ideas?

I think adding ``_()``  around the languages in tests/settings.py then a type assertion (``self.assertIsInstance(locale.__str__(), str)``) somewhere would be enough

**Comment by murray3k** [comment]
I added a test as suggested to the Locale model tests and the entries in `LANGUAGES` in tests/settings.py have the gettext `_()` applied.

Also in tests/test_utils.py I added the check `test_can_be_a_translation_proxy` to see if `get_content_languages` can handle a `_()` inside `WAGTAIL_CONTENT_LANGUAGES`. Might be unnecessary.

**Comment by zerolab** [comment]
Fixed in #6572
